test/functional/test_framework/test_node.py

```diff
@@ -293,8 +295,18 @@ def wait_for_cookie_credentials(self):
         self._raise_assertion_error("Unable to retrieve cookie credentials after {}s".format(self.rpc_timeout))
 
     def generate(self, nblocks, maxtries=1000000):
-        self.log.debug("TestNode.generate() dispatches `generate` call to `generatetoaddress`")
-        return self.generatetoaddress(nblocks=nblocks, address=self.get_deterministic_priv_key().address, maxtries=maxtries)
+        # if wallet is not compiled, blocks are generated to an address or descriptor using an HD seed
+        hd_address = self.getnewaddress()
```

Comment:
`getnewaddress` is a wallet RPC. If you want this to work when the wallet isn't compiled, you'll need to generate the HD keys in the python test framework.

Reply:
Thanks John! Would you mind giving an example of where this is done currently? I looked around in the test_framework and just the functional test dir in general, but I couldn't find any example of generating hd keys without wallet. Or maybe this is a function that doesn't exist in the functional test dir yet and we can write it?

Reply:
The RPC has non-wallet descriptor RPCs: https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md#support-for-output-descriptors-in-bitcoin-core

If I had to guess, you are probably looking for `deriveaddresses`

Reply:
I was thinking that the keys would be generated in the python test framework. I just assumed that we'd have some HD key generation code to test BIP32, but it appears I was wrong.

Marco's suggestion to use the `deriveaddress` seems ok, but I don't think you even need to do that. I think you can just pass the descriptor with keypath in to `generatetodescriptor`